repm/api.go

```diff
@@ -176,6 +213,73 @@ func (conn *connection) dispatchPullProgress(reqBytes []byte) ([]byte, error) {
 	return mustMarshal(res), nil
 }
 
+func (conn *connection) newTransaction() int {
+	conn.transactionMutex.Lock()
+	defer conn.transactionMutex.Unlock()
+	txID := conn.transactionCounter
+	conn.transactionCounter++
+	conn.transactions[txID] = conn.db.NewTransaction()
+	return txID
+}
+
+func (conn *connection) dispatchBeginTransaction(reqBytes []byte) ([]byte, error) {
+	var req BeginTransactionRequest
+	err := json.Unmarshal(reqBytes, &req)
+	if err != nil {
+		return nil, err
+	}
+
+	txID := conn.newTransaction()
+
+	res := BeginTransactionResponse{
+		TransactionID: txID,
+	}
+	return mustMarshal(res), nil
+}
+
+func (conn *connection) dispatchCloseTransaction(reqBytes []byte) ([]byte, error) {
+	var req TransactionRequest
```

Comment:
For documentation purposes let's just create a `BeginTransactionRequest` that is a renamed `TransactionRequest`

Reply:
Not sure what you want here?